translate/storage/catkeys.py

```diff
@@ -283,8 +299,47 @@ def parse(self, input):
     def serialize(self, out):
         output = csv.StringIO()
         writer = csv.DictWriter(output, FIELDNAMES, dialect="catkeys")
+        # Calculate/update fingerprint
+        self.header.checksum = self._compute_fingerprint()
         # No real headers, the first line contains metadata
         writer.writerow(dict(zip(FIELDNAMES, [self.header._header_dict[key] for key in FIELDNAMES_HEADER])))
         for unit in self.units:
             writer.writerow(unit.dict)
         out.write(output.getvalue().encode(self.encoding))
+
+    def _compute_fingerprint(self):
+        """
+        Compute the current hash key in the header for the current state of the store.
+
+        Inner function hashfun() based on CatKey::HashFun()
+        https://github.com/haiku/haiku/blob/b65adbdfbc322bb7d86d74049389c688e9962f15/src/kits/locale/HashMapCatalog.cpp#L93
```

Comment:
Nit: this last part of the docstring should be a docstring in the `hashfun` function below.